os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -0,0 +1,676 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ************************************************************************************/
+
+/************************************************************************************
+ * Included Files
+ ************************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+#include <debug.h>
+
+#include <tinyara/kmalloc.h>
+#include <tinyara/fs/ioctl.h>
+#include <tinyara/spi/spi.h>
+#include <tinyara/fs/mtd.h>
+
+#include "imxrt_flash.h"
+#include "imxrt_log.h"
+
+/************************************************************************************
+ * Pre-processor Definitions
+ ************************************************************************************/
+/* Various manufacturers may have produced the parts.  0x20 is the manufacturer ID
+ * for the STMicro MP25x serial FLASH.  If, for example, you are using the a Macronix
+ * International MX25 serial FLASH, the correct manufacturer ID would be 0xc2.
+ */
+
+#ifndef CONFIG_IMXRT_MANUFACTURER
+//#error  "Manufacturer Id not selected for the flash device"
+#endif
+
+#ifndef CONFIG_IMXRT_MEMORY_TYPE
+//#error  "Memory Type not defined no selected for the flash device"
+#endif
+
+#if defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT102x)
+/*  NorFlash capacity is 8,388,608 bytes:
+ *  (2048 sectors) * (4096 bytes per sector)
+ *  (32768 pages) * (256 bytes per page)
+ */
+#define IMXRT_1020EVK_NORFLASH_SECTOR_SHIFT    12	/* Sector size 1 << 12 = 4096 */
+#define IMXRT_1020EVK_NORFLASH_NSECTORS        2048
+#define IMXRT_1020EVK_NORFLASH_PAGE_SHIFT      8	/* Page size 1 << 8 = 256 */
+#define IMXRT_1020EVK_NORFLASH_NPAGES          32768
+#elif defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT105x)
+/*  HyperFlash capacity is 67,108,864 bytes:
+ *  (256 sectors) * (262,144 bytes per sector)
+ *  (131072 pages) * (512 bytes per page)
+ */
+#define IMXRT_1050EVK_HYPERFLASH_SECTOR_SHIFT    18	/* Sector size 1 << 18 = 262144 */
+#define IMXRT_1050EVK_HYPERFLASH_NSECTORS        256
+#define IMXRT_1050EVK_HYPERFLASH_PAGE_SHIFT      9	/* Page size 1 << 9 = 512 */
+#define IMXRT_1050EVK_HYPERFLASH_NPAGES          131072
+#else
+#endif
+
+/************************************************************************************
+ * Private Types
+ ************************************************************************************/
+
+/* This type represents the state of the MTD device.  The struct mtd_dev_s
+ * must appear at the beginning of the definition so that you can freely
+ * cast between pointers to struct mtd_dev_s and struct IMXRT_dev_s.
+ */
+
+struct imxrt_dev_s {
+	struct mtd_dev_s mtd;		/* MTD interface */
+	FAR struct spi_dev_s *dev;	/* Saved SPI interface instance */
```

Comment:
According to your mail, flexspi never affect to spi and flash using only flexspi, then you can remove spi_dev_s here.

Reply:
I fixed it